### PR TITLE
Get rid of monkey patched String.indent

### PR DESCRIPTION
**Pull request by JoeSSS**
Addresses #1546 

**Describe the solution you have implemented**
Moved indent method to console helpers instead on String patch.

### CODE REVIEW

**Comment by mattwynne** [comment]
Nice quick work!

Can I suggest we just stop extending `String` at all and make this a stand-alone helper function somewhere? i.e. delete that whole `core_ext` directory

**Comment by mattwynne** [comment]
i.e.

```
def indent(string, padding)
  ...
end
```

**Comment by JoeSSS** [comment]
Yeah, this is my plan :) just wanted to open a draft before 🛌

**Comment by mattwynne** [comment]
Fabulous!

**Comment by mattwynne** [comment]
Sorry for giving you feedback before you asked for it 😊

**Comment by JoeSSS** [comment]
> Sorry for giving you feedback before you asked for it 😊

no worries :) I think the basic implementation is done, now it is about fixing many tests, because indentation was off there 🙈

**Comment by JoeSSS** [comment]
I'm not sure I like the indentation of tags that I had to fix, but I do not have a hard opinion on it either.

**Comment by JoeSSS** [comment]
The current rpsec already covers the changes with tests.

**Comment by JoeSSS** [comment]
but I still added a couple of basic tests :)

**Comment by aurelien-reeves** [comment]
Great PR 🤩

Is it ready for review? Or do you still have some improvements to do?

**Comment by JoeSSS** [comment]
it is done from my point of view

**Comment by aurelien-reeves** [comment]
Ok.

Could you please update the changelog?

**Comment by JoeSSS** [comment]
Done, there is some infra issue with the tests

File: CHANGELOG.md
Comment (aurelien-reeves) [inline CHANGELOG.md:21]:
```suggestion
- Cucumber terminating with `negative argument` ArgumentError
  ([1546](https://github.com/cucumber/cucumber-ruby/issues/1546)
  [1548](https://github.com/cucumber/cucumber-ruby/pull/1548)
  [Serghei Moret](https://github.com/JoeSSS))
```

We usually just indicate the PR number

**Comment by aurelien-reeves** [comment]
Great, thank you :D

We'll wait for CircleCI to be fixed before approving and merging it 👍

**Comment by aurelien-reeves** [comment]
LGTM.

I'll merge this later this week to wait for other potential reviews

**Review by aurelien-reeves — APPROVED** [review]

**Review by luke-hill — COMMENTED** [review]
for arguments sake, it would be good to fix the method arg in this / the one below from indent to say indent_amount

**Comment by aurelien-reeves** [comment]
> 
> 
> for arguments sake, it would be good to fix the method arg in this / the one below from indent to say indent_amount

@luke-hill do you suggest the following?
```ruby
def indent(string, indent_amount)
```

Myself I have no issue with the argument being named "padding".

**Comment by luke-hill** [comment]
Sorry the method "invocation". It's a bit odd to see `indent(indent)`

File: lib/cucumber/formatter/pretty.rb
Comment (luke-hill) [inline lib/cucumber/formatter/pretty.rb:267]: @aurelien-reeves ie here I would have the method arg 2 renamed to `indent_amount` or `padding` if you like

Comment (aurelien-reeves) [reply lib/cucumber/formatter/pretty.rb:267]: Oh, ok, got it 👍

**Comment by JoeSSS** [comment]
hey, can it be merged, the `indent(indent)` thing is what was there before as well and not in the scope of the fix. But if it is really necessary I could take a look in a couple of weeks again 🌴

**Comment by luke-hill** [comment]
If you're busy @JoeSSS I'm happy to make the change as I'm working in here now.

**Comment by JoeSSS** [comment]
sure, added you to my fork. I will be on vacation and the plan is not to take the laptop with me :D

**Comment by JoeSSS** [comment]
thanks for the support!

**Comment by aslakhellesoy** [comment]
Hi @JoeSSS,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by mattwynne** [comment]
Awesome, thanks @JoeSSS!

If you're interested in helping us maintain Cucumber, you can find us in the #committers channel in the https://cucumber.io/community#slack

**Comment by olleolleolle** [comment]
@JoeSSS Hep! Nice to clean up the String class!

**Pull request merged**
